hoth_api.yaml

```diff
@@ -319,5 +319,351 @@ components:
+          maxItems: 4000
+        additional_info:
+          type: string
+          maxLength: 1000
+    EnvironmentInstance:
+      description: "Part of DOW, represents a single Environment Instance"
+      properties:
+        instance_name:
+          type: string
+          maxLength: 100
+        classification_level:
+          $ref: "#/components/schemas/ClassificationLevel"
+        instance_location:
+          type: string
+          enum:
+            - CSP
+            - Hybrid
+            - Not in the cloud / On-premise
+        csp_region:
+          $ref: "#/components/schemas/TargetCloudServiceProviderName" #On the SNOW side, this is CSP_REGION a choice field with no options, so using TargetCSPName instead
+        performance_tier: # This is also an empty choice on the SNOW side, set as empty String until we get details
+          type: string
+        pricing_model:
+          type: string
+          enum:
+            - On-demand
+            - Pay-as-you-go
+            - Reserved
+        pricing_model_expiration:
+          type: string
+          format: date
+          example: "2022-07-01"
+        operating_system_licensing:
+          type: string
+          maxLength: 1000
+        number_of_vCPUs:
+          type: integer
+          maximum: 40
+        storage_type:
+          type: string # Empty choice on SNOW side, set as empty String until we get details
+        storage_amount:
+          type: integer
+          maximum: 40
+        storage_unit:
+          $ref: "#/components/schemas/StorageUnit"
+        memory_amount:
+          type: integer
+          maximum: 40
+        memory_unit:
+          $ref: "#/components/schemas/StorageUnit"
+        data_egress_monthly_amount:
+          type: integer
+          maximum: 40
+        data_egress_monthly_unit:
+          $ref: "#/components/schemas/StorageUnit"
+    ClassificationLevel:
+      description: "Part of DOW, Classification Level table fields"
+      properties:
+        classification:
+          type: string
+          enum:
+            - Unclassified
+            - Top Secret
+            - Secret
+        impact_level:
+          type: string
+          nullable: true
+          enum:
+            - IL2
+            - IL4
+            - IL5
+            - IL6
+            - null
+    StorageUnit:
+      description: "Part of DOW, used in EnvironmentInstance schema"
+      type: string
+      enum:
+        - GB
+        - TB
+    PerformanceRequirements:
+      description: >
+        Part of DOW, encapsulates all data for 4. Performance Requirements and its subtasks. 
+        On the API side, the subtasks will filter through the provided required_services.service_offerings
+        to create the subtasks fields.
+      properties:
+        required_services:
+          type: array
+          items:
+            $ref: "#/components/schemas/RequiredService"
+          maxItems: 4000
```

Comment:
Does there need to be a `type` added here or above where the `$ref` is called? If so I think there are some `ref`s that need a `type` too.

Reply:
@juliusfitzhugh-ccpo Good call, although OpenAPI doesn't require a type, it is not inferred from the `properties` keyword. I remember this being an issue in the past when converting to JSON. I will update the spec to be explicit about `type: object`

Reply:
Slack overflow reference to what I was talking about: https://stackoverflow.com/questions/47374980/schema-object-without-a-type-attribute-in-swagger-2-0/47390723#47390723